src/python/src/grpc/_adapter/_channel.c

```diff
@@ -42,19 +42,35 @@
 static int pygrpc_channel_init(Channel *self, PyObject *args, PyObject *kwds) {
   const char *hostport;
   PyObject *client_credentials;
-  static char *kwlist[] = {"hostport", "client_credentials", NULL};
+  char *server_host_override;
```

Comment:
PyArg_ParseTupleAndKeywords ostensibly doesn't touch the C variable if it is an optional argument and Python user code does not specify the value. Might want to initialize this to `NULL`.